### PR TITLE
Add Bioconductor package scde.

### PR DESCRIPTION
**Pull request by jdblischak**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

Adds Bioconductor package [scde](https://www.bioconductor.org/packages/release/bioc/html/scde.html) for Single Cell Differential Expression. Recipe built with bioconda-utils 0.10.0.

### CODE REVIEW

**Comment by jdblischak** [comment]
I was able to build this recipe locally on a standard Linux machine, but the [mulled build failed](https://circleci.com/gh/bioconda/bioconda-recipes/3039?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) trying to load [Cairo](https://cran.r-project.org/package=Cairo):

```
[Feb 18 17:07:31] STEP Run image [quay.io/biocontainers/bioconductor-scde:2.6.0--r3.4.1_0] with command [[Rscript -e "library(scde)"]]
[Feb 18 17:07:31] DEBU Creating container [step-aded3cdba2]
[Feb 18 17:07:31] DEBU Created container [338c2754a3ab step-aded3cdba2], starting it
[Feb 18 17:07:32] DEBU Container [338c2754a3ab step-aded3cdba2] started, waiting for completion
[Feb 18 17:07:32] SERR Loading required package: flexmix
[Feb 18 17:07:32] SERR Loading required package: lattice
[Feb 18 17:07:32] SERR Error: package or namespace load failed for 'scde':
[Feb 18 17:07:32] SERR .onLoad failed in loadNamespace() for 'Cairo', details:
[Feb 18 17:07:32] SERR call: dyn.load(file, DLLpath = DLLpath, ...)
[Feb 18 17:07:32] SERR error: unable to load shared object '/usr/local/lib/R/library/Cairo/libs/Cairo.so':
[Feb 18 17:07:32] SERR libXrender.so.1: cannot open shared object file: No such file or directory
[Feb 18 17:07:32] SERR Execution halted
[Feb 18 17:07:33] ERRO Task processing failed: Unexpected exit code [1] of container [338c2754a3ab step-aded3cdba2], container preserved
```

**Comment by bgruening** [comment]
@jdblischak this means cairo is missing as runtime dep and your local test environment has cairo installed. Thats the nice feature of the mulled test, it can catch these things. However, if you can not get the dependency, you can use our extended base-docker image as a hack - e.g. as we do for opengl or UTF-8 stuff:

https://github.com/bioconda/bioconda-recipes/blob/2c0860ebea5e72f51479c8890db4ff6e3efe3e30/recipes/yahmm/meta.yaml#L40

**Comment by jdblischak** [comment]
> this means cairo is missing as runtime dep and your local test environment has cairo installed.

But the recipe specifies r-cairo as both a build and runtime dependency. And r-cairo in turn has cairo 1.14.* as both a build and runtime dependency.

https://github.com/conda-forge/r-cairo-feedstock/blob/ebf47277f12a9c084525a75e6a54bbc9d63aa9ff/recipe/meta.yaml#L40

In the build log, it says that cairo 1.14.10 is installed from conda-forge.

```
[Feb 18 17:07:16] SOUT cairo:                     1.14.10-0            conda-forge
```

Thus I'm at a loss as to how I can add the dependency if it is already there. Any ideas that I could experiment with?

**Comment by jdblischak** [comment]
I did some more investigating. Cairo is installed, but the mulled build is missing X11, hence the error:

```
libXrender.so.1: cannot open shared object file: No such file or directory
```

I found this [R help thread](http://r.789695.n4.nabble.com/Cairo-package-error-quot-unable-to-load-quot-td4694453.html) with a similar error, and the solution was to install X11. X11 isn't a dependency of Cairo, see https://github.com/conda-forge/cairo-feedstock/pull/23 for discussion.

I'll try adding [xorg-libx11](https://github.com/conda-forge/xorg-libx11-feedstock) as a dependency.

**Comment by bgruening** [comment]
@jdblischak maybe it is fixed with https://github.com/conda-forge/r-markdown-feedstock/pull/1 as well.
I'm running currently mulled to test this.

**Comment by jdblischak** [comment]
@bgruening That's the exact same Cairo/X11 error, so that looks like a better solution. I'll try adding `xorg-libxrender 0.9.10`. Thanks!

**Comment by bgruening** [comment]
Sure thing. I need to start building mulled containers for all r-* packages to catch those errors.

**Comment by jdblischak** [comment]
This time it failed due to a missing Qt library:

```
[Feb 23 22:44:26] SERR error: unable to load shared object '/usr/local/lib/R/library/Cairo/libs/Cairo.so':
[Feb 23 22:44:26] SERR libSM.so.6: cannot open shared object file: No such file or directory
```

Related SO question: https://stackoverflow.com/questions/32389599/anaconda-importerror-libsm-so-6-cannot-open-shared-object-file-no-such-file-o

**Comment by bgruening** [comment]
Mh, not sure, can you try xorg-libsm

**Comment by jdblischak** [comment]
The Linux build passed! But now macOS is failing due to `Library not loaded: @rpath/libc++.1.dylib`:

```
Error: package or namespace load failed for ‘scde’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/Users/distiller/project/miniconda/conda-bld/bioconductor-scde_1519434091680/_t_env/lib/R/library/scde/libs/scde.so':
  dlopen(/Users/distiller/project/miniconda/conda-bld/bioconductor-scde_1519434091680/_t_env/lib/R/library/scde/libs/scde.so, 6): Library not loaded: @rpath/libc++.1.dylib
```

From a quick search, it looks like this is related to llvm. It's already listed as a build dependency for osx.

File: recipes/bioconductor-scde/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    - r-nnet
+    - r-quantreg
+    - r-rcolorbrewer
+    - 'r-rcpp >=0.10.4'
+    - 'r-rcpparmadillo >=0.5.400.2.0'
+    - r-rjson
+    - r-rmtstat
+    - r-rook
+    - xorg-libxrender 0.9.10
+    - xorg-libsm
```
Comment (bgruening) [inline recipes/bioconductor-scde/meta.yaml]: I'm not sure but I think we should pin them as well?

**Comment by bgruening** [comment]
Its green now John.

**Comment by jdblischak** [comment]
@bgruening Awesome! And I've pinned the X11 dependencies as you did in https://github.com/conda-forge/r-rmarkdown-feedstock/pull/5/commits/2e35b271c392509c2c9f96e2630f2fca4f1424f1

**Comment by bgruening** [comment]
Thanks @jdblischak!

**Pull request merged**